test/apps/iodemo/io_demo.cc

```diff
@@ -2664,6 +2669,38 @@ static void adjust_opts(options_t *test_opts) {
         test_opts->chunk_size = std::min(test_opts->chunk_size,
                                          test_opts->max_data_size);
     }
+
+    if (servers.empty()) {
+        return;
+    }
+
+    for (auto server = servers.cbegin(); server != servers.cend(); ++server) {
+        repeat_separator = strchr(*server, ':');
+        if (repeat_separator == NULL) {
+            test_opts->servers.push_back(std::string(*server));
+            continue;
+        }
+
+        repeat_separator = strchr(repeat_separator + 1, ':');
+        if (repeat_separator == NULL) {
+            test_opts->servers.push_back(std::string(*server));
+            continue;
+        }
+
+        server_repeated = std::string(*server, repeat_separator - *server);
+        repeat_count    = atoi(repeat_separator + 1);
+        for (auto i = 0; i < repeat_count; i++) {
+            test_opts->servers.push_back(server_repeated);
+
+            if (test_opts->src_addrs.empty()) {
+                continue;
+            }
+
+            /* align with round robin method to select the src address */
+            test_opts->src_addrs.push_back(test_opts->src_addrs.at(
+                        std::distance(servers.cbegin(), server)));
```

Comment:
can't it exceed the size of src_addrs array?

Reply:
I'll fix the wrong logic here.

Reply:
Done. Logic has been removed after simplifying the design